micrometer-core/src/test/java/io/micrometer/core/instrument/binder/hystrix/MicrometerMetricsPublisherCommandTest.java

```diff
@@ -0,0 +1,152 @@
+
+    private static HystrixCommandGroupKey groupKey = HystrixCommandGroupKey.Factory.asKey("MicrometerGROUP");
+    private static HystrixCommandProperties.Setter propertiesSetter = HystrixCommandProperties.Setter()
+        .withCircuitBreakerEnabled(true)
+        .withCircuitBreakerRequestVolumeThreshold(20)
+        .withCircuitBreakerSleepWindowInMilliseconds(10_000)
+        .withExecutionIsolationStrategy(HystrixCommandProperties.ExecutionIsolationStrategy.THREAD)
+        .withExecutionTimeoutInMilliseconds(100)
+        .withMetricsRollingStatisticalWindowInMilliseconds(10_000)
+        .withMetricsRollingPercentileWindowInMilliseconds(1000)
+        .withMetricsRollingPercentileWindowBuckets(10);
+
+
+    @Test
+    public void testCumulativeCounters() throws Exception {
+        HystrixCommandKey key = HystrixCommandKey.Factory.asKey("MicrometerCOMMAND-A");
+        HystrixCommandProperties properties = new HystrixPropertiesCommandDefault(key, propertiesSetter);
+        HystrixCommandMetrics metrics = HystrixCommandMetrics.getInstance(key, groupKey, properties);
+        HystrixCircuitBreaker circuitBreaker = HystrixCircuitBreaker.Factory.getInstance(key, groupKey, properties, metrics);
+        SimpleMeterRegistry registry = new SimpleMeterRegistry();
+        MicrometerMetricsPublisherCommand metricsPublisherCommand = new MicrometerMetricsPublisherCommand(registry, key, groupKey, metrics, circuitBreaker, properties);
+        metricsPublisherCommand.initialize();
+
+        final int NUM_SECONDS = 5;
+
+        for (int i = 0; i < NUM_SECONDS; i++) {
+            new SuccessCommand(key).execute();
+            new SuccessCommand(key).execute();
+            new SuccessCommand(key).execute();
+            Thread.sleep(10);
+            new TimeoutCommand(key).execute();
+            new SuccessCommand(key).execute();
+            new FailureCommand(key).execute();
+            new FailureCommand(key).execute();
+            new SuccessCommand(key).execute();
+            new SuccessCommand(key).execute();
+            new SuccessCommand(key).execute();
+            Thread.sleep(10);
+            new SuccessCommand(key).execute();
+        }
+
+        Thread.sleep(500);
+
+        List<Tag> tags = Tags.zip("group", "MicrometerGROUP", "key", "MicrometerCOMMAND-A");
+        assertThat(registry.find("hystrix.execution").tags(Tags.concat(tags, "event", "success")).firstValue()).hasValue(40.0);
+        assertThat(registry.find("hystrix.execution").tags(Tags.concat(tags, "event", "timeout")).firstValue()).hasValue(5.0);
+        assertThat(registry.find("hystrix.execution").tags(Tags.concat(tags, "event", "failure")).firstValue()).hasValue(10.0);
+        assertThat(registry.find("hystrix.execution").tags(Tags.concat(tags, "event", "short_circuited")).firstValue()).hasValue(0.0);
+        assertThat(registry.find("hystrix.circuit.breaker.open").tags(tags).firstValue()).hasValue(0.0);
```

Comment:
Note the usage of the new `firstValue()` method to help leverage the `assertThat` syntax. I've been struggling with the `value(Statistic, value)` approach since it just returns an empty `Optional` if the value is wrong.

I prefer `assertThat` to complain, but report what it DID get as the value.

Reply:
I agree. We should expand the use of a technique like this in rc4